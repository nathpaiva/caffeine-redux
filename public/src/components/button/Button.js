import React from 'react';

var buttonStyle = {
  fontSize: '12px',
  padding: '10px',
  textAlign: 'center',
  textTransform: 'uppercase',
  cursor: 'pointer',
  backgroundColor: '#5A4A69',
  color: '#FFFFFF',
  fontWeight: 'bold',
  width: '100%',
  border: 'none',
  borderRadius: '2px',
};


const Button = ({ name }) => (
  <button style={buttonStyle}>{name}</button>
);

export default Button;
