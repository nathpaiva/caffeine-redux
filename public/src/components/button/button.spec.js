'use strict';
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import Button from './Button';

describe('#Button', () => {
  test('#should render a button without text', () => {
    const { container } = render(<Button />);

    expect(container.firstChild.tagName).toBe('BUTTON');
  });

  test('#should render a button with text', () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick}>Hello</Button>);

    const elementRendered = screen.getByText('Hello');
    expect(elementRendered).toBeInTheDocument();
    fireEvent.click(elementRendered);
    expect(onClick).toBeCalled();
  });
});
