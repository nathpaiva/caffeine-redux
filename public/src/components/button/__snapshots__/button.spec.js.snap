// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Test Button Component #Should Button to mach snapshot 1`] = `
<button
  style={
    Object {
      "backgroundColor": "#5A4A69",
      "border": "none",
      "borderRadius": "2px",
      "color": "#FFFFFF",
      "cursor": "pointer",
      "fontSize": "12px",
      "fontWeight": "bold",
      "padding": "10px",
      "textAlign": "center",
      "textTransform": "uppercase",
      "width": "100%",
    }
  }
>
  Entrar
</button>
`;
